test/cases/testdata/walkbuiltin/test-walkbuiltin-wildcard-path.yaml

```diff
@@ -0,0 +1,28 @@
+cases:
```

Comment:
For the sake of completeness, consider also creating a test case where we don't pass the output as an arg, but use the unification infix: e.g. `[_, value] = walk(obj)`. Don't think that makes much difference to the builtin, but it wouldn't hurt.

Reply:
could also throw in a

```rego
path = _
[path, value] = walk(a)
```

that might actually make it past the optimization. But if so, it's an edge case that I don't think warrants extra explanation in the builtin description.

Reply:
I checked using the [opa-explorer](https://github.com/srenatus/opa-explorer), and the compiler rewrites:

```rego
foo["bar"] {
    [path, value] := walk(input)
}
```
to
```rego
foo["bar"] { 
    __local3__ = input; walk(__local3__, __local2__)
    [__local0__, __local1__] = __local2__ 
}
```
So it'd be the same thing eventually either way. Other tests seem to cover that since before.